apis/cloud.redhat.com/v1alpha1/clowdapp_types.go

```diff
@@ -378,3 +384,43 @@ func Labels(labels map[string]string) omfunc {
 		o.SetLabels(labels)
 	}
 }
+
+func GetAppInSameEnv(client client.Client, ctx context.Context, app *ClowdApp, appList *ClowdAppList) error {
+	err := client.List(ctx, appList)
+
+	if err != nil {
+		return errors.New("Could not get app list")
+	}
+
+	for _, iapp := range appList.Items {
+
+		if iapp.Spec.Pods != nil {
+			iapp.ConvertToNewShim()
+		}
+
+		if iapp.Spec.EnvName != app.Spec.EnvName {
+			continue
+		}
+		appList.Items = append(appList.Items, iapp)
+	}
+	return nil
+}
+
+func GetAppForDBInSameEnv(client client.Client, ctx context.Context, app *ClowdApp) (*ClowdApp, error) {
+	appList := &ClowdAppList{}
+	var refApp ClowdApp
+
+	err := GetAppInSameEnv(client, ctx, app, appList)
```

Comment:
could you use `client.MatchingFields` here to combine both the `EnvName` and `Name` checks?